content/blog/2025-05-13-SCIF.md

```diff
@@ -0,0 +1,116 @@
++++
+title = "Implementing Multiple Contracts and Error Messaging For a Compiler for Smart Contracts"
+[extra]
+bio = """
+  Noah Schiff is a just-graduated undergrad / CS MEng student interested in applied programming languages and optimizing software engineering.<br>
+  Kabir Samsi is a third-year undergrad, interested in building programming languages and compilers that can 
+  target new areas.br>
+  Stephanie Ma is a first-year MS student interested in PL and compilers.<br>
+"""
+latex=true
+[[extra.authors]]
+name = "Noah Schiff"
+[[extra.authors]]
+name = "Kabir Samsi"
+[[extra.authors]]
+name = "Stephanie Ma"
++++
+
+## Background
+
+Our project focuses on improving and adding new features to the compiler for [Scif](https://arxiv.org/pdf/2407.01204) – a language for representing smart contracts with secure control flow. SCIF as a programmming language uses information flow and its type system to help to prevent control-flow attacks and improve improve secure smart contracts.
+
+As both a language design and implementation paper, the [SCIF technical report](https://arxiv.org/abs/2407.01204) extensively discusses the SCIF compiler and the correctness and performance of the Solidity code it generates for SCIF programs. As such, its authors hope to eventually publish the compiler as a research artifact similar to what is required for a [PLDI Research Artifact](https://pldi25.sigplan.org/track/pldi-2025-pldi-research-artifacts) or [OOPSLA Artifact](https://2025.splashcon.org/track/splash-2025-oopsla-artifacts). The [ACM discusses different badges](https://www.acm.org/publications/policies/artifact-review-and-badging-current) an artifact can be awarded. The goal for SCIF is that the compiler can be easily setup and run by any open source contributor and that the compiler can easily validate results described in papers.
+
+In our project, we focused on a few primary aspects – adding on the much-desired feature of defining **multiple contracts in one file**, and allowing this to work with our the compiler's current control flow and functionality; improving the quality of error messaging for malformed files; and improving the structure of the compiler's build system.
```

Comment:
> improving the quality of error messaging for malformed files

Unless I somehow missed it, the rest of your post does not discuss this. Please either omit it from this list (if you did not actually do it) or add a section about it.